diff --git a/data/asserts.info b/data/asserts.info
new file mode 100644
index 0000000000..7a855ed70f
--- /dev/null
+++ b/data/asserts.info
@@ -0,0 +1 @@
+SNAPD_ASSERTS_FORMATS='{"snap-declaration":5,"system-user":1}'
\ No newline at end of file
diff --git a/mkversion.sh b/mkversion.sh
index 62f41c0167..4e4d87798a 100755
--- a/mkversion.sh
+++ b/mkversion.sh
@@ -137,7 +137,11 @@ if [ "$GO111MODULE" = "off" ] ; then
 elif [ ! -d "$GO_GENERATE_BUILDDIR/vendor/github.com"  ] ; then
     MOD=--
 fi
-fmts=$(cd "$GO_GENERATE_BUILDDIR" ; go run $MOD ./asserts/info)
+if [ ! -f ./data/asserts.info ]; then
+    fmts=$(cd "$GO_GENERATE_BUILDDIR" ; go run $MOD ./asserts/info)
+else
+    fmts=$(cat ./data/asserts.info)
+fi
 
 cat <<EOF > "$PKG_BUILDDIR/data/info"
 VERSION=$v
 